tox.ini

```
@@ -0,0 +1,63 @@
+# Generated from:
+# https://github.com/zopefoundation/meta/tree/master/config/pure-python
+[tox]
+envlist =
+    lint
+    py27
+    py35
+    py36
+    py37
+    py38
+    py39
+    coverage
+
+[testenv]
+usedevelop = true
+deps =
+    zope.testrunner
+commands =
+    zope-testrunner --test-path=src []
+extras =
+    test
+
+[testenv:lint]
+basepython = python3
+skip_install = true
+deps =
+    flake8
+    check-manifest
+    check-python-versions
+commands =
+    flake8 src setup.py
+    check-manifest
+    check-python-versions
+
+[testenv:coverage]
+basepython = python3
+deps =
+    coverage
+    coverage-python-version
+    zope.testrunner
+commands =
+    coverage run -m zope.testrunner --test-path=src []
+    coverage html
+    coverage report -m --fail-under=83
+
+[coverage:run]
+branch = True
+plugins = coverage_python_version
+source = src
```

Ideally this would be

```suggestion
source = cipher.longrequest
```

so we don't have to rely on `usedevelop = true`.

It's probably not worth changing now.

---

You are right, but I still do not like to be forced to run tox in development mode to get paths in tracebacks pointing to the source code. I've been burned several times changing the files in the distribution created for tox.